File: src/main/scala/org/cloudio/morpheus/mail/traditional/EmployeeAdapter.scala
package org.cloudio.morpheus.mail.traditional

import java.util.Date

import org.cloudio.morpheus.mail.MailOwner

/**
 * Created by zslajchrt on 24/08/15.
 */
trait EmployeeAdapter extends MailOwner {
  this: Employee =>

  override def nick(): String = employeeCode

  override def email(): String = employeeCode + "@bigcompany.com"

  override def birthDate(): Date = birth
}

File: src/main/java/org/cloudio/morpheus/mail/EmployeeAdapter.java
package org.cloudio.morpheus.mail;

import java.util.Date;

/**
 * Created by zslajchrt on 24/08/15.
 */
public class EmployeeAdapter extends Employee implements MailOwner {
    public EmployeeAdapter() {
    }

    public EmployeeAdapter(Employee other) {
        super(other);
    }

    public String nick() {
        return getEmployeeCode();
    }

    public String firstName() {
        return getFirstName();
    }

    public String lastName() {
        return getLastName();
    }

    public String email() {
        return getEmployeeCode() + "@bigcompany.com";
    }

    public Date birthDate() {
        return getBirth();
    }
}
